src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/parser/GenericCodeCommentParser.java

```diff
@@ -0,0 +1,96 @@
+package de.uhd.ifi.se.decision.management.jira.extraction.parser;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import de.uhd.ifi.se.decision.management.jira.model.git.ChangedFile;
+import de.uhd.ifi.se.decision.management.jira.model.git.CodeComment;
+
+/**
+ * Extracts decision knowledge elements from code comments.
+ */
+public class GenericCodeCommentParser {
+	private List<CodeComment> commentList = new ArrayList<CodeComment>();
+
+	public List<CodeComment> getComments(ChangedFile inspectedFile) {
+		if (this.commentList.size() == 0) {
+			String fileContent = inspectedFile.getFileContent() + "\n";
```

Comment:
There are a lot of if-else statements in this method. Can it be split into smaller methods/simplified?